File: instantlib/src/main/java/com/baselib/instant/observer/state/RegisterState.java
package com.baselib.instant.observer.state;


import com.baselib.instant.util.LogUtils;

/**
 * 注册状态下对解注进行响应
 *
 * @author wsb
 */
public class RegisterState extends BaseRegisterState {
    @Override
    public boolean unregister(IRegisterStateOperate registerStateOperate) {
        LogUtils.i(this + "现进行广播解注");
        return registerStateOperate.realUnregister();
    }
}

File: instantlib/src/main/java/com/baselib/instant/observer/state/RegisterState.kt
package com.baselib.instant.observer.state


import com.baselib.instant.util.LogUtils

/**
 * 注册状态下对解注进行响应
 *
 * @author wsb
 */
class RegisterState : BaseRegisterState() {
    override fun unregister(registerStateOperate: IRegisterStateOperate): Boolean {
        LogUtils.i(this.toString() + "现进行广播解注")
        return registerStateOperate.realUnregister()
    }
}
